bgpd/bgp_clist.c

```diff
@@ -1123,12 +1125,28 @@ static bool lcommunity_list_valid(const char *community)
 		frrstr_split(communities[j], ":", &splits, &num);
 
 		for (int i = 0; i < num; i++) {
-			if (strtoul(splits[i], NULL, 10) > UINT_MAX)
-				return false;
-
 			if (strlen(splits[i]) == 0)
+				/* There is no digit to check */
```

Comment:
This is not related to this change, but we need to make sure we don't leak the memory in every return. An example here: https://github.com/FRRouting/frr/pull/7803/commits/0e8916e0cffcc5558dc7659f09752156785ea7d3#diff-844b6a37729199284c5f42cdc4ce195d3b4d5e90f1b781549c837d4bf0a79c0dR664

Reply:
Oh, good point. I'll address this and include with my fix for the failing i386 ubuntu topotest

Reply:
Followed your linked example and applied the same technique to prevent the memory leak that was present here